src/test/java/org/snomed/snowstorm/core/data/services/TraceabilityLogServiceTest.java

```diff
@@ -644,6 +651,86 @@ void testRebaseWithChangeSupersededByOtherProject() throws Exception {
 		assertTrue(componentChange.isSuperseded());
 	}
 
+	@Test
+	void testExtensionUpgradingDependencyWithSameComponentEdited() throws ServiceException, InterruptedException {
+		Concept concept;
+		CodeSystem codeSystem;
+		String extMain = "MAIN/SNOMEDCT-TEST";
+		Relationship relationship;
+
+		// Create International Concepts
+		concept = conceptService.create(
+				new Concept().addFSN("Vehicle (vehicle)").addRelationship(ISA, Concepts.SNOMEDCT_ROOT),
+				MAIN
+		);
+		String vehicleId = concept.getConceptId();
+		concept = conceptService.create(
+				// Published with wrong modelling
+				new Concept().addFSN("Car (vehicle)").addRelationship(ISA, Concepts.SNOMEDCT_ROOT),
+				MAIN
+		);
+		String carId = concept.getConceptId();
+		String relationshipToBeInactivated = concept.getRelationships().iterator().next().getRelationshipId();
+
+		// Version International
+		codeSystem = codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT", MAIN));
+		codeSystemService.createVersion(codeSystem, 20220131, "20220131");
+
+		// Create Extension
+		codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT-TEST", extMain));
+		concept = conceptService.create(
+				new Concept().addFSN("Extension module (core metadata concept)").addRelationship(ISA, Concepts.MODULE),
+				extMain
+		);
+		String extModule = concept.getConceptId();
+		branchService.updateMetadata(extMain, Map.of(Config.DEFAULT_MODULE_ID_KEY, extModule));
+
+		// International fixes modelling (through classification)
+		concept = conceptService.find(carId, MAIN);
+		concept.getRelationships().iterator().next().setActive(false).updateEffectiveTime();
+		concept.getRelationships().add(new Relationship(ISA, vehicleId));
+		concept = conceptService.update(concept, MAIN);
+		List<String> intRelationships = getRelationshipIds(concept);
+
+		// Version International
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220731, "20220731");
+
+		// Extension fixes modelling (through classification)
+		concept = conceptService.find(carId, extMain);
+		concept.getRelationships().iterator().next().setActive(false).updateEffectiveTime();
+		concept.getRelationships().add(new Relationship(ISA, vehicleId));
+		concept = conceptService.update(concept, extMain);
+		List<String> extRelationships = getRelationshipIds(concept);
```

Comment:
@dmcgihtsdo  Good test. But I think the traceability assertion for inactivation should be here before upgrade.

Reply:
Mm, how do you mean @CoderMChu? 

Before the upgrade, the Extension will inactivate which goes through as INACTIVATE. During the upgrade, Snowstorm identifies there is a duplicate edit and discards the Extension's change. This eventually leads to the traceability message being DELETE instead of INACTIVATE.

So I need the assertion after the upgrade?

Reply:
@dmcgihtsdo This bug is about wrong change type used when extension inactivates international content. This can happen during classification for relationships. In your test case the inactivation happens before upgrade, that's why I added a comment. Your assumption about "During the upgrade, Snowstorm identifies there is a duplicate edit and discards the Extension's change. This eventually leads to the traceability message being DELETE instead of INACTIVATE." is not correct. Discarding changes during rebase will only lead to changes to be superseded. The change type of a traceability is determined by the nature of change for a given component. This can happen for activity type "CONTENT_CHANGE" or "CLASSIFICATION_SAVE". If you are still not clear we can have a quick call.